apps/mobile/src/libs/components/categories-form/categories-form.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+
+import { Button, Checkbox, Text, View } from "~/libs/components/components";
+import { NumericValues } from "~/libs/enums/enums";
+import { useAppForm, useCallback, useFormController } from "~/libs/hooks/hooks";
+import { toastMessage } from "~/libs/packages/toast-message/toast-message";
+import { globalStyles } from "~/libs/styles/styles";
+import {
+	type CategoryDto,
+	quizCategoriesValidationSchema,
+} from "~/packages/quiz/quiz";
+
+import { defaultCategoriesFormValue } from "./libs/values/values";
+import { styles } from "./styles";
+
+type Properties = {
+	categories: CategoryDto[];
+};
+
+const CategoriesForm: React.FC<Properties> = ({ categories }) => {
+	const { control, errors, handleSubmit } = useAppForm({
+		defaultValues: defaultCategoriesFormValue,
+		validationSchema: quizCategoriesValidationSchema,
+	});
+
+	const { field } = useFormController({
+		control,
+		name: "categories",
+	});
+
+	const { onChange, value: selectedCategories } = field;
+
+	const handleOnSubmit = useCallback((): void => {
```

Comment:
onSubmit should come as a prop

Reply:
Not fixed

Reply:
Okay, could you please explain in more detail what you meant? Thank you.